### PR TITLE
Add google translate enc converter

### PR DESCRIPTION
**Pull request by bobeaton**
added another Translator converter for Google Translate (and fixed a bug in the backtranslation control for when there are multiple translators displaying at the same time)

### CODE REVIEW

**Comment by papeh** [comment]
> doesn't change the state of the control if the parent control isn't visible (which it isn't initially)

If this is what the FLEx team sometimes sees, it could be solved by
```
protected override void OnLoad(EventArgs) {
  // do stuff
  base.OnLoad(e);
}
```

File: src/BackTranslationHelper/BackTranslationHelperCtrl.cs [hunk 1]
```diff
@@ -829,8 +829,16 @@ private void TargetTextToolStripMenuItem_Click(object sender, EventArgs e)
                     textBoxPossibleTargetTranslation2.Font =
                     textBoxPossibleTargetTranslation3.Font =
                     textBoxTargetBackTranslation.Font = fontDialog.Font;
-                mapProjectNameToTargetFontOverride.Add(projectName, fontOverride);
-                Properties.Settings.Default.MapProjectNameToTargetFontOverride = SettingFromDictionary(mapProjectNameToTargetFontOverride);
+
+				if (mapProjectNameToTargetFontOverride.ContainsKey(projectName))
+				{
+					mapProjectNameToTargetFontOverride[projectName] = fontOverride;
```
Comment (papeh) [inline src/BackTranslationHelper/BackTranslationHelperCtrl.cs:835]: `dict[key] = value` should work whether or not the key exists. (no need for a the surrounding if statement)

File: src/EcTranslators/DeepLTranslator/DeepLTranslatorEncConverter.cs [hunk 2]
```diff
@@ -27,10 +27,10 @@ namespace SilEncConverters40.EcTranslators.DeepLTranslator
 	public class DeepLTranslatorEncConverter : EncConverter
     {
 		#region Const Definitions
-		// by putting the azure key in a settings file, users can get their own free azure account, create their own 'Translator'
-		//  resource, and enter their own key in the file (or the UI to have us set it in the file) and get their own 2E6 chars free
+		// by putting the DeepL key in a settings file, users can get their own free DeepL account, create their own 'Translator'
+		//  resource, and enter their own key in the file (or the UI to have us set it in the file) and get their own 500K chars free
 		//	(or pay for as much as they want)
-		// see https://docs.microsoft.com/en-us/azure/cognitive-services/translator/quickstart-translator
```
Comment (papeh) [inline src/EcTranslators/DeepLTranslator/DeepLTranslatorEncConverter.cs:33]: Good catch! Thanks for keeping comments updated.

File: src/EcTranslators/GoogleTranslator/GoogleTranslatorAutoConfigDialog.cs [hunk 3]
```diff
@@ -483,80 +281,47 @@ private void comboBoxTargetLanguages_SelectedIndexChanged(object sender, EventAr
...
+			// only send the key if it's already the override key (so we don't expose ours)
+			var translatorCredentialsOverride = Properties.Settings.Default.GoogleTranslatorCredentialsOverride;
+			if (!String.IsNullOrEmpty(translatorCredentialsOverride))
+				translatorCredentialsOverride = EncryptionClass.Decrypt(translatorCredentialsOverride);
 
-			// we don't initialize the script language(s) based on the source language changing unless we're in the Transliterate mode
-			string selectedItem = (string)comboBoxSourceLanguages.SelectedItem;
-			if ((SourceLanguageNameAutoDetect == selectedItem) ||
-				(transductionSelected != TransductionType.Transliterate))
+			var dlg = new QueryForGoogleCredentials(translatorCredentialsOverride);
```
Comment (papeh) [inline src/EcTranslators/GoogleTranslator/GoogleTranslatorAutoConfigDialog.cs:308]: `using (var dlg = new . . .)` to ensure `dlg` is properly disposed

File: src/EcTranslators/app.config [hunk 4]
```diff
@@ -70,6 +42,63 @@
...
...1bbHdm5dcnydMH2JoOcSUtK99S/ilE9TpKgjiBK4BzXIkXY+BkSyIRAdJc0lnyL/O8uwnc9u+4orUm0e+7oZdHpMa6OovpC/md6r0nOhMgsN04VYEynpimqirAai2yZliAlnl/+AXIshwqrJdDxekARvCkHYb2WRR0PDpSSpdVEHug/uC6xhEM7XBTrymz14fUS22y7T+XMue0PCzrBpdySw/EbjEWvgBR21XrDT03Gt7ZLiiFW4tWq4foAgeJAkPEGb+qKucblC+xydpONoCvzY0GiyzTdhPK110vSt0zT+cCYEy1L21gnRXiABuftJNi/+P9OqrmgvK9xIvOS/katiDGviXpD9jvVpyhIjOZLxOQtslzEAps5JpyRARISO0pttcY0hWT9znSTL2Oi/7GxRulx0OJnUjC1qg21HKxVVmXd4IObRbH6bSLDDObkmnyeyM5ZUUnQMgA7LnaZYJqxGI415hTjqmpX0vMn7wwo7R3+NkfccpVVmofGCRHjgumw3xriALsztBFNJf3DjFlvxMPTgxabjvj1Xy/HipTz+vEcAEr48Bfg7t+JTJEa2LGy/G3wAMDMbEyU7zFJL3vfC2mJUlgj7qdq/1quIpwkjnyIq6UVaEfPs/dSHzKYodrMoxFdTLBrbl86FhlXHyjlHcHImOddRALgBgjtRFzlfjG5B0DvmwJ7D8f0wanP8l/g0xzLf9RgKdU2evUBIlu0y1hYfT5wD0YLP2rqsMs/TcS4YLAvEwoP0Zmo4LBnCWajSyJmm0jxvkeulEnJ0hsyslqEY6XBfWgzS61mJKZNZes0JbjIptjiUXBW4RzAfYe+o91fpnjOqp+Nfv8cDno1vpWHkkEHT+HX/IVTBDqsMqmEsl2kFNgSIVwZpBT4E+wrFGciFjiiCU10A6On6kbfvn6FKbCK1CKvWIcuuufQXq6xyF67OVtvciTl22EslIHiDLjTMTYL4rg3Pk4fsovYBanwhf6dlhUxcZ+W6RkurPz/lEJ4ucPR2ifaLnVIubj9jGk6o63TuHrGThpXKomxBLgVcWBxhkY8XHF6+Z/qhYbYAuGW8OgblOT/8Pk/Y0NFCcv5Rmj52NzWHjFGxj0Znsg4OnaurMKfeNnU5PbwyCn2z1GFII0SnMbXAo3gWVxHxJzvq7UjlVfMFVC8yz+ZzoReS5HDZJxwSzmDQ</value>
```
Comment (papeh) [inline src/EcTranslators/app.config:91]: is this value needed both here and in settings.settings?

**Review by papeh — APPROVED** [review]

File: src/BackTranslationHelper/BackTranslationHelperCtrl.cs (hunk 1 above)
Comment (bobeaton) [reply src/BackTranslationHelper/BackTranslationHelperCtrl.cs:835]: thanks. changed them all

File: redist/Help/Bing_Translator_Plug-in_About_box_files/image003.jpg
[the remark this answers is not in the document]
Comment (bobeaton) [reply redist/Help/Bing_Translator_Plug-in_About_box_files/image003.jpg:1]: good catch, Apparently, if you edit the *.htm file in Word and resize an image, it creates a new image leaving the old one behind and making it effectively incompatible w/ the htm file in a browser...

File: src/EcTranslators/Common.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
+using ECInterfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SilEncConverters40.EcTranslators
+{
+	public class Common
```
[the remark this answers is not in the document]
Comment (bobeaton) [reply src/EcTranslators/Common.cs:10]: changed the class to internal and renamed to EcTranslatorUtils

File: src/EcTranslators/GoogleTranslator/GoogleTranslatorAutoConfigDialog.cs (hunk 3 above)
Comment (bobeaton) [reply src/EcTranslators/GoogleTranslator/GoogleTranslatorAutoConfigDialog.cs:308]: thanks. done

File: src/EcTranslators/Properties/Settings.settings [hunk 6]
```diff
@@ -14,17 +14,29 @@
...
+    </Setting>
+    <Setting Name="GoogleTranslatorKeyDialogInstructionUrl" Type="System.String" Scope="User">
+      <Value Profile="(Default)">https://cloud.google.com/translate/docs/setup#creating_service_accounts_and_keys</Value>
```
[the remark this answers is not in the document]
Comment (bobeaton) [reply src/EcTranslators/Properties/Settings.settings:33]: yes!

File: src/EcTranslators/app.config (hunk 4 above)
Comment (bobeaton) [reply src/EcTranslators/app.config:91]: apparently, adding something to the Settings, adds it automatically to the app.config file, bkz the latter eventually becomes the <assemblyname>.config file and is installed on client machines. So yes.

**Review by papeh — APPROVED** [review]

**Pull request merged**